File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let message (logLine: string): string =
    let msg = logLine.Split(':')[1]
    msg.Trim()

let logLevel(logLine: string): string =
    let level = logLine.Split(':')[0]
    level.Replace("[", "").Replace("]", "").ToLower().Trim()

let reformat(logLine: string): string =
    let msg = message logLine
    let level = logLevel logLine
    $"{msg} ({level})"

File: solutions/csharp/log-levels/1/LogLevels.cs
using System;
using System.Linq;

static class LogLine
{
    public static string Message(string logLine)
    {
        return logLine.Split(':').Last().Trim();
    }

    public static string LogLevel(string logLine)
    {
        return logLine.Split('[').Last().Split(']').First().ToLower();
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}
